support/cas-server-support-ldap-monitor/src/main/java/org/apereo/cas/monitor/config/LdapMonitorConfiguration.java

```diff
@@ -36,16 +40,33 @@
     private CasConfigurationProperties casProperties;
 
     @Bean
+    ListFactoryBean pooledLdapConnectionFactoryHealthIndicatorFactoryBean() {
+        return new ListFactoryBean() {
+            @Override
+            protected void destroyInstance(final List list) {
+                list.forEach(connectionFactory -> {
+                    ((ConnectionFactory) connectionFactory).close();
+                });
+            }
+        };
+    }
+
+    @Bean
+    @SneakyThrows
+    @Autowired
     @ConditionalOnEnabledHealthIndicator("pooledLdapConnectionFactoryHealthIndicator")
-    public CompositeHealthContributor pooledLdapConnectionFactoryHealthIndicator() {
+    public CompositeHealthContributor pooledLdapConnectionFactoryHealthIndicator(
+            final ListFactoryBean pooledLdapConnectionFactoryHealthIndicatorFactoryBean) {
```

Comment:
Specify a qualifier for the injected parameter by its name, to avoid conflicts in the future